src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -143,11 +150,33 @@ public Set<AccessProfile> generateAccessProfilesByCaseReference(String caseRefer
         FilteredRoleAssignments filteredRoleAssignments =
             roleAssignmentsFilteringService.filter(roleAssignments, caseDetails.get());
 
-
         CaseTypeDefinition caseTypeDefinition = caseDefinitionRepository.getCaseType(caseDetails.get().getCaseTypeId());
 
-        return Sets.newHashSet(filteredAccessProfiles(filteredRoleAssignments.getFilteredMatchingRoleAssignments(),
-            caseTypeDefinition, false));
+        List<AccessProfile> filteredAccessProfiles = filteredAccessProfiles(
+            filteredRoleAssignments.getFilteredMatchingRoleAssignments(), caseTypeDefinition, false);
+
+        final String caseAccessCategory = getCaseAccessCategoryFromCaseData(caseDetails.get());
+
+        return filterAccessProfilesByCaseAccessCategory(filteredAccessProfiles, caseAccessCategory);
+    }
+
+    private Set<AccessProfile> filterAccessProfilesByCaseAccessCategory(List<AccessProfile> filteredAccessProfiles,
+                                                                        final String caseAccessCategory) {
+        return filteredAccessProfiles.stream()
+            .filter(fap -> fap.getCaseAccessCategories() == null
+                || (!Strings.isEmpty(caseAccessCategory)
+                && Arrays.stream(fap.getCaseAccessCategories().split(","))
+                .map(String::trim)
+                .anyMatch(s -> s.startsWith(caseAccessCategory))))
```

Comment:
I think this is the wrong way round?
```suggestion
                .anyMatch(s -> caseAccessCategory.startsWith(s))))
```
It should pass through if the value on the case starts with the value of the access profile.

Reply:
done